ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -637,8 +640,38 @@ runThreadNetwork systemTime ThreadNetworkArgs
         ledger <- atomically $ ledgerState <$> getExtLedger
         -- Combine the node's seed with the current slot number, to make sure
         -- we generate different transactions in each slot.
-        let txs = runGen (nodeSeed `combineWith` unSlotNo curSlotNo) $
-                    testGenTxs numCoreNodes curSlotNo cfg txGenExtra ledger
+        let genTxs i =
+                runGen
+                  (nodeSeed `combineWith` unSlotNo curSlotNo `combineWith` i)
+                  (testGenTxs numCoreNodes curSlotNo cfg txGenExtra ledger)
+
+        -- These TxGen generators fail by invoking 'QC.discard'. We try many
+        -- times before actually giving up. When we give up, it fatal, not just
+        -- a discard.
+        let loop acc = do
```

Comment:
I had the impression `acc` was an accumulator of transactions, but it's actually more of a seed, right? Maybe a different name and type signature would clarify it.

Reply:
Done!